armadillo/src/main/java/org/molgenis/armadillo/controller/StorageController.java

```diff
@@ -0,0 +1,292 @@
+import static org.molgenis.armadillo.audit.AuditEventPublisher.UPLOAD_OBJECT;
+import static org.springframework.http.HttpHeaders.CONTENT_DISPOSITION;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM;
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM_VALUE;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE;
+import static org.springframework.http.ResponseEntity.noContent;
+import static org.springframework.http.ResponseEntity.notFound;
+import static org.springframework.web.bind.annotation.RequestMethod.HEAD;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+import javax.validation.Valid;
+import org.molgenis.armadillo.audit.AuditEventPublisher;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.molgenis.armadillo.storage.ArmadilloStorageService;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@Tag(name = "storage", description = "API to manipulate the storage")
+@RestController
+@Valid
+@SecurityRequirement(name = "http")
+@SecurityRequirement(name = "bearerAuth")
+@SecurityRequirement(name = "JSESSIONID")
+@RequestMapping("storage")
+public class StorageController {
+
+  private final ArmadilloStorageService storage;
+  private final AuditEventPublisher auditor;
+
+  public StorageController(ArmadilloStorageService storage, AuditEventPublisher auditor) {
+    this.storage = storage;
+    this.auditor = auditor;
+  }
+
+  @Operation(summary = "List all projects")
```

Comment:
I would propose to remove this one in favour of the /admin/projects that has more information. What is use case to duplicate?

Reply:
Ah yes, good point. It didn't consider removing it yet. The endpoint does make a lot of "REST" sense, but it might confuse people about which API to use. Removing 👍